src/lib.rs

```diff
@@ -56,11 +56,28 @@ impl Cli {
 
         let delete_file = run::generate_delete_file(tmp_dir.path())?;
 
-        run::start_shell(config, tmp_dir.path())?;
+        let subprocesses = run::start_subprocesses(config, tmp_dir.path());
 
-        run::clean_up(delete_file, tmp_dir, self.worktree_branch.clone())?;
+        let res = run::start_shell(config, tmp_dir.path());
 
-        Ok(())
+        #[cfg(windows)]
+        if config.editor.is_some() {
+            unsafe {
+                windows_sys::Win32::System::Console::FreeConsole();
+            }
+        }
```

Comment:
You're probably freeing the console too late. I doubt this works as expected because start_shell will wait for the process to exit.